### ISSUE TITLE
Diagnostics should be mergeable without requiring a separate Merged variation class.

### ISSUE THREAD

**Issue by benshayden** [catapult-project/catapult#3607]
Currently, some Diagnostics have singular fields, do not themselves support merging, and only support merging via a MergedFoo variation class: MergedTelemetryInfo, MergedBuildbotInfo, MergedDeviceInfo, MergedRevisionInfo.

This situation makes it difficult to consume the data in these diagnostics. Consumers must use type checks:
```javascript
const telemetryInfo = tr.v.d.TelemetryInfo.getFromHistogram(hist);
if (telemetryInfo instanceof tr.v.d.MergedTelemetryInfo) // consume plural Set fields like labels and benchmarkNames
else // consume singular fields like label and benchmarkName
```

CSVBuilder and HistogramSet.GROUPINGS were broken when the Merged variations were introduced because they were not updated to use type-checks like this.

In order to prevent bugs whereby consumers fail to type-check, the variations with singular fields should be deprecated and removed in favor of always using the plural variations, which should be renamed to remove the Merged prefix.
Some callers may need to test whether the Set fields are empty, contain a single element, or contain multiple elements, but these should be fewer, simpler, and less error-prone than the current type-checks.

Relatedly, RelatedEventSet should also be replaced by CollectedRelatedEventSet, which should be renamed to RelatedEventSet, but that is blocked on plumbing canonical trace url through telemetry: #2431 

There may be some way to reframe the semantics of Generic and CollectedGeneric such that the two classes can be merged into one class. GenericSet maybe?

@eakuefner
